openshift/templates/api-dc-template.yaml

```diff
@@ -5,99 +5,101 @@ metadata:
-          restartPolicy: Always
-          terminationGracePeriodSeconds: 30
-          dnsPolicy: ClusterFirst
-          securityContext: {}
-          schedulerName: default-scheduler
+- kind: DeploymentConfig
+  apiVersion: apps.openshift.io/v1
+  metadata:
+    annotations:
+      app.openshift.io/route-disabled: 'false'
+    name: ${APPLICATION_NAME}
+    namespace: ${LICENSE_PLATE}-${ENVIRONMENT}
+    labels:
+      app: ${APPLICATION_NAME}
+      app.kubernetes.io/component: ${APPLICATION_NAME}
+      app.kubernetes.io/instance: ${APPLICATION_NAME}
+      app.kubernetes.io/name: ${APPLICATION_NAME}
+      app.kubernetes.io/part-of: coco-demo
+      app.openshift.io/runtime: nodejs
+      app.openshift.io/runtime-namespace: ${LICENSE_PLATE}-${ENVIRONMENT}
+  spec:  
+    strategy:
+      type: Rolling
+      rollingParams:
+        updatePeriodSeconds: 1
+        intervalSeconds: 1
+        timeoutSeconds: 600
+        maxUnavailable: 25%
+        maxSurge: 25%
+      resources: {}
+      activeDeadlineSeconds: 21600
+    triggers:
+      - type: ConfigChange
+    replicas: 1
+    revisionHistoryLimit: 10
+    test: false
+    selector:
+      app: ${APPLICATION_NAME}
+      deploymentconfig: ${APPLICATION_NAME}
+    template:
+      metadata:
+        creationTimestamp: null
+        labels:
+          app: ${APPLICATION_NAME}
+          deploymentconfig: ${APPLICATION_NAME}
+        annotations:
+          openshift.io/generated-by: OpenShiftWebConsole
+          vault.hashicorp.com/agent-inject: 'true'
+          vault.hashicorp.com/agent-inject-token: 'false' 
+          vault.hashicorp.com/agent-pre-populate-only: 'true' 
+          vault.hashicorp.com/auth-path: 'auth/k8s-silver'
+          vault.hashicorp.com/namespace: 'platform-services'
+          vault.hashicorp.com/role: ${VAULT_ENVIRONMENT}
+          vault.hashicorp.com/agent-inject-secret-coco-secrets-${ENVIRONMENT}: ${VAULT_ENVIRONMENT}/${ENVIRONMENT}/sso-integration
+          vault.hashicorp.com/agent-inject-template-coco-secrets-${ENVIRONMENT}: |
+            {{- with secret "${VAULT_ENVIRONMENT}/${ENVIRONMENT}/sso-integration" }}  
+            export SSO_AUTH_SERVER_URL="{{ .Data.data.SSO_AUTH_SERVER_URL }}"
+            export SSO_CLIENT_ID="{{ .Data.data.SSO_CLIENT_ID }}"
+            export SSO_CLIENT_SECRET="{{ .Data.data.SSO_CLIENT_SECRET }}"
+            export SSO_ENVIRONMENT="{{ .Data.data.SSO_ENVIRONMENT }}"
+            export SSO_INTEGRATION_ID="{{ .Data.data.SSO_INTEGRATION_ID }}"
+            {{- end }}
+      spec:
+        serviceAccountName: baf95e-vault
+        containers:
+          - name: ${APPLICATION_NAME}
+            image: artifacts.developer.gov.bc.ca/tec1-citz-imb-coco/coco-api:${IMAGE_TAG}
+            imagePullSecrets:
+            - name: artifactory-creds
+            ports:
+              - containerPort: 6600
+                protocol: TCP
+            env:
+              - name: ENVIRONMENT
+                value: prod
```

Comment:
```suggestion
                value: "${ENVIRONMENT}"
```

Reply:
this would make the value dev for dev and test for test.

is that what you want @BradyMitch